src/App.tsx

```diff
@@ -1,8 +1,16 @@
-import React from 'react';
+import { useState } from 'react';
 import 'bulma/css/bulma.css';
 import './App.scss';
+import cn from 'classnames';
 
-export const goodsFromServer = [
+type GoodsFromServer = {
+  id: number;
+  val: string;
+};
+
+type GoodsList = GoodsFromServer[];
+
+const goodsFromServer: string[] = [
```

Comment:
The `goodsFromServer` array should contain objects with `id` and `value` properties to match the `GoodsFromServer` type. Currently, it's an array of strings, which doesn't align with the expected type.